File: 2570-merge-two-2d-arrays-by-summing-values/2570-merge-two-2d-arrays-by-summing-values.java
class Solution {
    public int[][] mergeArrays(int[][] nums1, int[][] nums2) {
        int n1=nums1.length,n2=nums2.length,i=0,j=0;
        ArrayList<int []>ans=new ArrayList();
        while(i<n1&& j<n2){
            if(nums1[i][0]==nums2[j][0]){
                ans.add(new int[]{nums1[i][0],nums1[i][1]+nums2[j][1]});
                i++;
                j++;
            }
            else if(nums1[i][0]<nums2[j][0]){
                ans.add(nums1[i]);
                i++;
            }
            else{
                ans.add(nums2[j]);
                j++;
            }
        }
        while(i<n1){
            ans.add(nums1[i++]);
        }
        while(j<n2){
            ans.add(nums2[j++]);
        }
        return ans.toArray(new int[ans.size()][]);
    }
}

File: 2570-merge-two-2d-arrays-by-summing-values/2570-merge-two-2d-arrays-by-summing-values.cpp
class Solution {
public:
    vector<vector<int>> mergeArrays(vector<vector<int>>& nums1, vector<vector<int>>& nums2) {
        vector<vector<int>>ans;
        int temp1=0, temp2=0;
        int n1 = nums1.size(), n2 = nums2.size();
        while(temp1<n1 && temp2<n2){
            if(nums1[temp1][0]==nums2[temp2][0]){
                ans.push_back({nums1[temp1][0],nums1[temp1][1]+nums2[temp2][1]});
                temp1++;temp2++;
            }
            else if(nums1[temp1][0]<nums2[temp2][0]){
                ans.push_back({nums1[temp1][0],nums1[temp1][1]});
                temp1++;               
            }
            else{
                ans.push_back({nums2[temp2][0],nums2[temp2][1]});
                temp2++;                
            }
        }

        while(temp1<n1){
            ans.push_back({nums1[temp1][0],nums1[temp1][1]});
            temp1++; 
        }

        while(temp2<n2){
            ans.push_back({nums2[temp2][0],nums2[temp2][1]});
            temp2++; 
        }
        return ans; 
    }
};